libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -391,6 +391,10 @@ method onTopicSubscription*(p: PubSub, topic: string, subscribed: bool) {.base.}
   # Called when subscribe is called the first time for a topic or unsubscribe
   # removes the last handler
 
+  # Check that this is an allowed topic
+  if p.subscriptionValidator != nil and p.subscriptionValidator(topic) == false:
+    warn "Trying to subscribe to a topic not passing validation!", topic
```

Comment:
arguably we shouldn't add the topic to the subscription table at all in this case because it leads to further protocol violations and descoring when the others subscribe/graft